src/python/pants/engine/internals/parametrize.py

```diff
@@ -111,14 +168,67 @@ def expand(
                 {field_name: alias for field_name, alias, _ in parametrized_args}
             )
             parametrized_args_fields = tuple(
-                (field_name, field_value) for field_name, _, field_value in parametrized_args
+                (field_name, field_value)
+                for field_name, _, field_value in parametrized_args
+                # Exclude any parametrize group
+                if not (isinstance(field_value, Parametrize) and field_value.is_group)
             )
             expanded_fields: dict[str, Any] = dict(non_parametrized + parametrized_args_fields)
+            expanded_fields.update(
+                # There will be at most one group per cross product.
+                next(
+                    (
+                        field_value.kwargs
+                        for _, _, field_value in parametrized_args
+                        if isinstance(field_value, Parametrize) and field_value.is_group
+                    ),
+                    {},
+                )
+            )
             yield expanded_address, expanded_fields
 
+    @staticmethod
+    def _collect_parametrizations(
+        fields: dict[str, Any | Parametrize]
+    ) -> Mapping[str | None, list[tuple[str, Parametrize]]]:
+        parametrizations = defaultdict(list)
+        for field_name, v in fields.items():
+            if not isinstance(v, Parametrize):
+                continue
+            group_name = None if not v.is_group else v.group_name
+            parametrizations[group_name].append((field_name, v))
+        return parametrizations
+
+    @staticmethod
+    def _check_parametrizations(
+        parametrizations: Mapping[str | None, list[tuple[str, Parametrize]]]
+    ) -> None:
+        for group_name, groups in parametrizations.items():
+            if group_name is None:
+                continue
+            if len(groups) > 1:
+                raise ValueError(f"Parametrization group name is not unique: {group_name!r}")
```

Comment:
Does this mean overriding the group from `__defaults__` no longer works?

Reply:
For now, yes. I think we want to address that.

Reply:
Done.